### DISCUSSION TITLE
is there any method or attribute which is used to prevent the extraction of watermark, page number or background text

### DISCUSSION THREAD

**Discussion by hemilparmar** [py-pdf/pypdf#1255] [Q&A]
when i try to extract some unwanted text is also coming so want to ask you do pypdf has any method or attribute to prevent those kind of extraction. please answer this.

**Comment by pubpub-zz** [comment]
You can limit text extraction to some orientation:
https://pypdf2.readthedocs.io/en/latest/user/extract-text.html

This could help for water marks if they are rotated other criteria are not yet implemented
(https://github.com/py-pdf/PyPDF2/discussions/1181#discussioncomment-3413144)
